app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,65 @@
+class MiqCockpitWsWorker::Runner < MiqWorker::Runner
+  BINDING_ADDRESS = ENV['BINDING_ADDRESS'] || (Rails.env.production? ? nil : "0.0.0.0")
+
+  def self.start_worker(*args)
+    runner = MiqCockpitWsWorker::Runner.new(*args)
+
+    runner.prepare
+    runner.start_cockpit_ws
+    runner.run
+  end
+
+  def do_work
+  end
+
+  def pid_file
+    Rails.root.join("tmp/pids/cockpit-ws.pid")
+  end
+
+  def delete_pid_file
+    File.delete(pid_file) if File.exist?(pid_file)
+  end
+
+  def heartbeat(*args)
+    ret = super
+    if !@cockpit_pid || !MiqProcess.alive?(@cockpit_pid)
+      @cockpit_pid = nil
+      do_exit("Cockpit-ws not running, exiting!", 1)
+    end
+    ret
```

Comment:
rubyrep has since been deleted because we switched over to a different replication engine.  However the worker code for doing process management was very stable.  We would have to resurrect it from history.